pkg/keyring/algo/ed25519.go

```diff
@@ -0,0 +1,54 @@
+package algo
+
+import (
+	"github.com/cosmos/cosmos-sdk/crypto/types"
```

Comment:
This comes from the previous PR (https://github.com/smartcontractkit/chainlink-terra/pull/45) that conflated protocol level transaction signing with OCR2 signing. We want to avoid Terra/Cosmos specific libraries here and use `crypto/ed25519` + `hdevalence/ed25519consensus`

Reply:
The Cosmos SDK has the best Keyring lib implementation that we found. Their `keyring` package has a nice API and allows us to hook in custom storage backends and different signing algorithms functionality as required by new integrations.

We use the Cosmos SDK secp256k1 implementation for Terra and Solana, but like @archseer is suggesting we should use `crypto/ed25519` + `hdevalence/ed25519consensus` instead Cosmos SDK ed25519 impl.

There are a [couple of backend storage options available](https://github.com/cosmos/cosmos-sdk/blob/86474748888204515f59aaeab9be295066563f46/crypto/keyring/keyring.go#L31), but we'd like to add an SQL option as familiar to our NodeOps.

Reply:
- We didn't want to write one ourselves from scratch
- We didn't want to use Core's `services/keyring`:
  - It's not really a lib or designed to be used as such, but more procedural code for current core usecases
  - We needed to keep Solana & Terra quiet so PRs to Core's repo would be inappropriate
  - We want to keep all network-specific code in separate monorepos to avoid excess context switching, so adding something like `ocrterrakey` (consequence of point #1) in the core repo is not the way to go
  

So the solution we propose:

- Cosmos SDK `crypto/keyring` package as the underlying library
- Relay SDK `keyring` package uses the Cosmos SDK and adds additional capabilities we need (SQL backend, ed25519 signing)
- Any Relay implementation project can add its own custom backends or signing algos, if not supported upstream